src/polar/PolarGrid.tsx

```diff
@@ -18,159 +18,146 @@ interface PolarGridProps {
+  };
 
-  /**
-   * Draw concentric circles
-   * @param {Number} radius The radius of circle
-   * @param {Number} index  The index of circle
-   * @param {Object} extraProps Extra props
-   * @return {ReactElement} circle
-   */
-  renderConcentricCircle(radius: number, index: number, extraProps?: SVGProps<SVGCircleElement>) {
-    const { cx, cy } = this.props;
-    const props = {
-      stroke: '#ccc',
-      ...filterProps(this.props),
-      fill: 'none',
-      ...extraProps,
-    };
-
-    return (
-      <circle
-        {...props}
-        className="recharts-polar-grid-concentric-circle"
-        key={`circle-${index}`}
-        cx={cx}
-        cy={cy}
-        r={radius}
-      />
-    );
-  }
+  return (
+    <circle
+      {...concentricCircleProps}
+      className="recharts-polar-grid-concentric-circle"
+      key={`circle-${index}`}
+      cx={cx}
+      cy={cy}
+      r={radius}
+    />
+  );
+};
+
+// Draw concentric polygons
+const ConcentricPolygon: React.FC<ConcentricProps> = props => {
+  const { radius, index } = props;
+  const concentricPolygonProps = {
+    stroke: '#ccc',
+    ...filterProps(props),
+    fill: 'none',
+  };
 
-  /**
-   * Draw concentric polygons
-   * @param {Number} radius     The radius of polygon
-   * @param {Number} index      The index of polygon
-   * @param {Object} extraProps Extra props
-   * @return {ReactElement} polygon
-   */
-  renderConcentricPolygon(radius: number, index: number, extraProps?: SVGProps<SVGPathElement>) {
-    const props = {
-      stroke: '#ccc',
-      ...filterProps(this.props),
-      fill: 'none',
-      ...extraProps,
-    };
-
-    return (
-      <path
-        {...props}
-        className="recharts-polar-grid-concentric-polygon"
-        key={`path-${index}`}
-        d={this.getPolygonPath(radius)}
-      />
-    );
+  return (
+    <path
+      {...concentricPolygonProps}
+      className="recharts-polar-grid-concentric-polygon"
+      key={`path-${index}`}
+      d={getPolygonPath(radius, props.cx, props.cy, props.polarAngles)}
+    />
+  );
+};
+
+// Draw concentric axis
+// TODO Optimize the name
+const ConcentricPath: React.FC<Props> = props => {
+  const { polarRadius, gridType } = props;
+
+  if (!polarRadius || !polarRadius.length) {
+    return null;
   }
 
-  /**
-   * Draw concentric axis
-   * @return {ReactElement} Concentric axis
-   * @todo Optimize the name
-   */
-  renderConcentricPath() {
-    const { polarRadius, gridType } = this.props;
-
-    if (!polarRadius || !polarRadius.length) {
-      return null;
-    }
-
-    return (
-      <g className="recharts-polar-grid-concentric">
-        {polarRadius.map((entry: number, i: number) =>
-          gridType === 'circle' ? this.renderConcentricCircle(entry, i) : this.renderConcentricPolygon(entry, i),
-        )}
-      </g>
-    );
+  return (
+    <g className="recharts-polar-grid-concentric">
+      {polarRadius.map((entry: number, i: number) => {
+        const key = i;
```

Comment:
nit: why assign a variable to `i` which is already defined?

Reply:
Actually just to trick ESLint regarding the `no-array-index-key` rule, that otherwise shows a warning.